### PR TITLE
spec: VolumeHandle replaces VolumeID, VolumeMetadata

### PR DESCRIPTION
**Pull request by jdef**
Refactor volume ID and metadata; provide additional clarification.
xref #29 #88

### CODE REVIEW

**Comment by jdef** [comment]
xref #71

File: spec.md [hunk 1]
```diff
@@ -537,35 +537,33 @@ message VolumeInfo {
...
+// that each Plugin keeps the information herein as small as possible.
+message VolumeHandle {
+
+  // ID is the identity of the provisioned volume specified by the
+  // Plugin. This field is REQUIRED. 
+  // This information SHALL NOT be considered sensitive such that, for
+  // example, the CO MAY generate log messages that include this data.
+  string id = 1;
+
+  // Metadata captures additional, possibly sensitive information about
```
Comment (saad-ali) [inline spec.md]: nit: `additional, possibly sensitive, information`

File: spec.md [hunk 2]
```diff
@@ -537,35 +537,33 @@ message VolumeInfo {
...
-  // each Plugin keeps this information as small as possible.
-  map<string, string> values = 1;
+// VolumeHandle objects are generated by Plugin implementations to
+// serve two purposes: first, to efficiently identify a volume known
+// to the Plugin implementation; second, to capture additional, opaque,
+// possibly sensitive information associated with a volume. It SHALL
+// NOT change over the lifetime of the volume and MAY be safely cached
+// by the CO.
+// Since this object will be passed around by the CO, it is RECOMMENDED
+// that each Plugin keeps the information herein as small as possible.
```
Comment (saad-ali) [inline spec.md]: Maybe add in a size requirement, like we did with credentials. Always better to start with a hard limit and loosen it in the future, then not starting with a limit and trying to add it later. e.g. `The total bytes of the ID and metadata must be less than 1 Mebibyte.`

File: spec.md (hunk 1 above)
Comment (julian-hj) [reply spec.md]: I am not sure i agree with this nit.  To me it reads better without the extra comma.

Comment (saad-ali) [reply spec.md]: Having a single comma suggests that "Metadata captures additional" is an introductory clause. But, "possibly sensitive information about..." can't stand on it's on (it is incomplete without the first part of the sentence).

Instead, "possibly sensitive" is nonessential information. And the sentence reads find without it.

From https://owl.english.purdue.edu/owl/owlprint/607/:

```
Commas with Nonessential Elements

Some modifying elements of a sentence are essential, restricting the meaning of a modified term, while others are nonessential and don't restrict the modified term's meaning. These nonessential elements, which can be words, phrases, or clauses, are set off with commas.

Rule: Use commas before and after nonessential words, phrases, and clauses, that is, elements embedded in the sentence that interrupt it without changing the essential meaning.
```

So I would offset "possibly sensitive" with two commas.

I'm being very pedantic here, so feel free to disregard =)

File: spec.md [hunk 3]
```diff
@@ -537,35 +537,33 @@ message VolumeInfo {
...
+message VolumeHandle {
+
```
Comment (jieyu) [inline spec.md]: nit: kill this line to be consistent with other places.

File: spec.md (hunk 2 above)
Comment (jdef) [reply spec.md]: agreed

File: spec.md (hunk 1 above)
Comment (jdef) [reply spec.md]: ack

File: spec.md (hunk 3 above)
Comment (jdef) [reply spec.md]: ack

File: spec.md (hunk 2 above)
Comment (jdef) [reply spec.md]: xref #87

**Comment by saad-ali** [comment]
LGTM. Thanks @jdef. Merging.

**Pull request merged**